cmake/version.cmake

```diff
@@ -1,20 +1,43 @@
-set(PROJECT_VERSION_MAJOR 1)
-set(PROJECT_VERSION_MINOR 9)
-set(PROJECT_VERSION_PATCH 99)
-set(PROJECT_VERSION_TWEAK 0)
+include($ENV{ZEPHYR_BASE}/cmake/hex.cmake)
+file(READ $ENV{ZEPHYR_BASE}/VERSION ver)
+
+string(REGEX MATCH "VERSION_MAJOR = ([0-9]*)" _ ${ver})
+set(PROJECT_VERSION_MAJOR ${CMAKE_MATCH_1})
+
+string(REGEX MATCH "VERSION_MINOR = ([0-9]*)" _ ${ver})
+set(PROJECT_VERSION_MINOR ${CMAKE_MATCH_1})
+
+string(REGEX MATCH "PATCHLEVEL = ([0-9]*)" _ ${ver})
+set(PROJECT_VERSION_PATCH ${CMAKE_MATCH_1})
+
+string(REGEX MATCH "VERSION_TWEAK = ([0-9]*)" _ ${ver})
+set(PROJECT_VERSION_TWEAK ${CMAKE_MATCH_1})
+
+string(REGEX MATCH "EXTRAVERSION = (.*)" _ ${ver})
+set(PROJECT_VERSION_EXTRA ${CMAKE_MATCH_1})
+
+
+message("Zephyr version: ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
```

Comment:
Maybe message(STATUS? The rest of the code uses the mode keyword. Also looks like a stray newline.